### PR TITLE
add doctest support

### LINKED ISSUE #4317: **Run** should run doctests
Relevant code:

* https://github.com/rust-analyzer/rust-analyzer/blob/d1c1c01309cff76f6ba60c845500f67a1ca6cec6/crates/ra_ide/src/runnables.rs#L43 -- here we decide if there's a test under cursor
* https://github.com/rust-analyzer/rust-analyzer/blob/d1c1c01309cff76f6ba60c845500f67a1ca6cec6/crates/rust-analyzer/src/main_loop/handlers.rs#L377 -- here we convert "test under cursor" into json we send to LSP client
*  https://github.com/rust-analyzer/rust-analyzer/blob/d1c1c01309cff76f6ba60c845500f67a1ca6cec6/editors/code/src/commands/runnables.ts#L9 -- here we run the actual tests

### PR DESCRIPTION
**Pull request by bnjjj**
close #4317

preview:

Capture d’écran 2020-05-05 à 17 39 58

### CODE REVIEW

File: crates/rust-analyzer/src/main_loop/handlers.rs [hunk 1]
```diff
@@ -835,6 +835,7 @@ pub fn handle_code_lens(
     for runnable in world.analysis().runnables(file_id)? {
         let title = match &runnable.kind {
             RunnableKind::Test { .. } | RunnableKind::TestMod { .. } => "▶️\u{fe0e}Run Test",
+            RunnableKind::DocTest { .. } => "▶️\u{fe0e}Run Doctest",
```
Comment (bjorn3) [inline crates/rust-analyzer/src/main_loop/handlers.rs:838]: When a doc comment has multiple tests this should say `Run doctests`.

Comment (matklad) [reply crates/rust-analyzer/src/main_loop/handlers.rs:838]: I think its fine :)

**Comment by matklad** [comment]
bors r+

I think we should also handle doctests on things other than functions, but this is a good start!

File: crates/rust-analyzer/src/main_loop/handlers.rs (hunk 1 above)
Comment (bnjjj) [reply crates/rust-analyzer/src/main_loop/handlers.rs:838]: If you change your mind I already implemented the version with trailing `s` when multiple doc tests. Let me know if you're interested :)

**Pull request merged**